src/NHibernate/Collection/Generic/PersistentGenericMap.cs

```diff
@@ -347,8 +358,7 @@ public ICollection<TValue> Values
 		{
 			get
 			{
-				Read();
-				return WrappedMap.Values;
+				return _wrappedValues;
```

Comment:
`Read()` moved to `ValuesWrapper` `ICollection<TValue>` members. Otherwise, reaching `Values` for calling `AsQueryable` was loading the map.